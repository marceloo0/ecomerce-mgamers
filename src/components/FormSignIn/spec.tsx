import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'

import { FormSignIn } from '.'

describe('<FormSignIn />', () => {
  it('should render default', () => {
    const { container } = renderWithTheme(<FormSignIn />)

    expect(screen.getByText(/login/i)).toBeInTheDocument()
    expect(screen.getByText(/Acesse com conta Google/i)).toBeInTheDocument()
    expect(screen.getByText(/Acesse com conta GitHub/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
