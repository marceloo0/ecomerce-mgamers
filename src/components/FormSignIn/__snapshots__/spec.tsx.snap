// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignIn /> should render default 1`] = `
.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 1px solid transparent;
  border-radius: 0.4rem;
  cursor: pointer;
  color: #C7ECEC;
  background-color: #201C2B;
  padding: 1rem;
}

.c8:hover {
  background-color: #332756;
}

.c8 > svg {
  color: #7FFA45;
  margin-right: 0.5rem;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.5rem 1rem;
  background-color: #201C2B;
  z-index: 99;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
}

.c2 {
  border: none;
  background-color: transparent;
}

.c3 {
  width: 6rem;
}

.c4 {
  height: 100%;
  width: 100%;
  z-index: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  font-size: 3.8rem;
  font-weight: 600;
  color: #2E2F42;
}

.c0 {
  background-color: #201C2B;
  height: 100vh;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 15rem;
}

.c7 {
  color: #C7ECEC;
}

@media screen and (min-width:768px) and (max-width:1023px) {
  .c3 {
    width: 7rem;
  }
}

@media (min-width:1024px) {
  .c3 {
    width: 9rem;
  }
}

@media screen and (min-width:768px) and (max-width:1023px) {
  .c5 {
    font-size: 7rem;
  }
}

@media (min-width:1024px) {
  .c5 {
    font-size: 10rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <button
      class="c2"
    >
      <img
        alt="logo"
        class="c3"
        role="img"
        src="/img/logo.svg"
      />
    </button>
  </div>
  <div
    class=""
  >
    <div
      class="c4"
    >
      <span
        class="c5"
      >
        Login
      </span>
    </div>
  </div>
  <div
    class="c6"
  >
    <h3
      class="c7"
    >
      Escolha o que melhor lhe atende
    </h3>
    <button
      class="c8"
    >
      <svg
        enable-background="new 0 0 48 48"
        fill="currentColor"
        height="36"
        stroke="currentColor"
        stroke-width="0"
        version="1.1"
        viewBox="0 0 48 48"
        width="36"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <path
          d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12
	c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24
	c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
          fill="#FFC107"
        />
        <path
          d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657
	C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
          fill="#FF3D00"
        />
        <path
          d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36
	c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
          fill="#4CAF50"
        />
        <path
          d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571
	c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
          fill="#1976D2"
        />
      </svg>
      <span>
        Acesse com conta Google
      </span>
    </button>
    <button
      class="c8"
    >
      <svg
        fill="currentColor"
        height="36"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 496 512"
        width="36"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
        />
      </svg>
      <span>
        Acesse com conta GitHub
      </span>
    </button>
  </div>
</div>
`;
